apps/gallery/js/startup.js

```diff
@@ -261,23 +257,34 @@
     var placeholder = $('thumbnails-placeholder');
     placeholder.parentElement.replaceChild(thumbnails, placeholder);
 
-    // Handle clicks on the thumbnails
-    thumbnails.addEventListener('click', thumbnailClickHandler);
-
     // When the first page of thumbnails is diplayed, we can emit
     // our 'visually complete' mark for startup time comparison
     Thumbnails.firstpage.then(function() {
       // Tell performance monitors that "above the fold" content is displayed
-      // and is ready to interact with.
       window.performance.mark('visuallyLoaded');
-      window.performance.mark('contentInteractive');
 
       // Remove the attribute to trigger gaia header font-fit logic, defer it
-      // til now to postpone the overhead.
-      var headers = document.querySelectorAll('gaia-header');
-      for (var i = 0; i < headers.length; ++i) {
-        headers[i].removeAttribute('no-font-fit');
-      }
+      // til now to postpone the overhead. Add setTimeout to lazyload
+      // gaia-header.js aynchronously.
+      setTimeout(function() {
+        LazyLoader.load(['shared/elements/gaia-header/dist/gaia-header.js'],
+          function() {
+            var headers = document.querySelectorAll('gaia-header');
+            for (var i = 0; i < headers.length; ++i) {
+              headers[i].removeAttribute('no-font-fit');
+            }
+          });
+      });
+
+      // Clicking on the select button goes to thumbnail select mode
+      $('thumbnails-select-button').onclick =
+        setView.bind(null, LAYOUT_MODE.select);
+
+      // Handle clicks on the thumbnails
+      thumbnails.addEventListener('click', thumbnailClickHandler);
+      // Tell performance monitors that "above the fold" content
+      // is ready to interact with.
+      window.performance.mark('contentInteractive');
```

Comment:
On rethinking gallery app content interactive mark on app launch should be all thumbnail items on first page are interactive which we are ensuring, so I will prefer not to move 'contentInteractive' mark and keep the code as it was.